controllers/complianceeventsapi/server.go

```diff
@@ -528,17 +558,82 @@ func getSingleComplianceEvent(db *sql.DB, w http.ResponseWriter, r *http.Request
 	}
 }
 
+func getClusterNameFromID(ctx context.Context, db *sql.DB, clusterID string) (name string) {
+	err := db.QueryRowContext(ctx,
+		`SELECT name FROM clusters WHERE cluster_id = $1`, clusterID,
+	).Scan(&name)
+	if err != nil {
+		return ""
+	}
+
+	return name
+}
+
 // getWhereClause will convert the input queryOptions to a WHERE statement and return the filter values for a prepared
 // statement.
-func getWhereClause(options *queryOptions) (string, []any) {
+func getWhereClause(ctx context.Context, options *queryOptions,
+	db *sql.DB, config *rest.Config, token string, client client.Client,
+) (string, []any, error) {
 	filterSQL := []string{}
 	filterValues := []any{}
 
+	allManagedClusterRules, err := getAllManagedClusterRules(ctx, config, token, client)
+	if err != nil {
+		return "", nil, err
+	}
+
+	var filterClusterNames []string
+
+	// "True" means the user added "cluster_id" or "cluster.name" as url filters
+	hasClusterFilter := false
+
 	for sqlColumn, values := range options.Filters {
 		if len(values) == 0 {
 			continue
 		}
 
+		// "cluster" change to "clusters" at queryOptionsToSQL
+		if sqlColumn == "clusters.cluster_id" {
+			for _, value := range values {
+				clusterName := getClusterNameFromID(ctx, db, value)
+
+				rules, ok := allManagedClusterRules[clusterName]
+				if !ok {
+					continue
+				}
+
+				if slices.Contains(rules, "get") || slices.Contains(rules, "*") {
+					filterClusterNames = append(filterClusterNames, "'"+clusterName+"'")
+				} else {
+					log.Info("The service account does not have GET authorization for " + clusterName)
+				}
+			}
+
+			hasClusterFilter = true
+
+			continue
+		}
+
+		// "cluster" change to "clusters" at queryOptionsToSQL
+		if sqlColumn == "clusters.name" {
+			for _, value := range values {
+				rules, ok := allManagedClusterRules[value]
+				if !ok {
+					continue
+				}
+
+				if slices.Contains(rules, "get") || slices.Contains(rules, "*") {
+					filterClusterNames = append(filterClusterNames, "'"+value+"'")
```

Comment:
This could be susceptible to SQL injection. It's unlikely now because your current code does validate the cluster names and I don't think Kubernetes would allow such special characters in cluster names, but this check will be removed after my review comments are addressed. Could you please use the prepared statement style of using `$1, $2` and etc like the rest of the code? Then we take out any risk.